### ISSUE TITLE
Uninstalling 1.8 tools does not remove some its "features"

### ISSUE THREAD

**Issue by Vaccano** [Microsoft/TypeScript#7527]
I did a fresh build of my machine yesterday (from base OS).

I Installed Visual Studio 2015 Update 1.  Got my project from source and got it compiling great.

Today I installed the Typescript 1.8 tools (Said 1.8.4 I think).  I got this from the Visual Studio 2015 link on the Typescript 1.8 blog post.

Once I loaded my project it asked me if I wanted to upgrade my tools.  I said yes.

I then build again, and got a ton of errors.  I could not see any reason for them, so I decided to downgrade back to 1.7 tools.

I uninstalled Typescript 1.8 tools and reverted my project back.

When I started up visual studio again my project loaded fine but gave me a build error.  The error was complaining about "compilerOptions":being null or undefined. (Which was odd, because it was defined in my tsconfig file.)

**Error:**  
I went and looked in my project properties and it told me that because I had a tsconfig file, the properties were disabled.

**Expected:**  
Because I had uninstalled the 1.8 tools, my project would go back to ignoring the tsconfig file.

**Workaround**  
I was able to get around this by removing my tsconfig file from my project.

**Comment by mhegazy** [comment]
> I then build again, and got a ton of errors.

what kind of errors did you get?

**Comment by Vaccano** [comment]
I got a whole bunch of "Cannot find name ...." errors.

Everything from $ to aurelia-framework.

It works fine with the older version.

**Comment by Vaccano** [comment]
@mhegazy So I tried it again, to see if I could get you a list of errors.  I noticed that (at least this time) all the errors only show when I have "Build + Intellisense" selected in the dropdown in the "Error List".

If I just pick "Build" then there are no errors.

But the red squiglies on all my imports will drive me nuts.   Plus it has kill my type intellisense completion.

**Comment by mhegazy** [comment]
I think i know what is going on. in TS 1.7 the project file was used to build your project, so the tsconfig.json was ignored. in TS 1.8 the tsconfig.json is now honored. looks like your tsconfig.json does not match what you had in your project file. try looking in the build log (or output window after build), what arguments does `tsc.exe` takes, and make sure your tsconfig.json has these set as well.

**Comment by Vaccano** [comment]
@mhegazy But I uninstalled the TS 1.8 tools.  And I changed my project file to use the 1.7 tools again.

I guess I thought that would revert me back to TS 1.7.  Is there more that I need to uninstall to revert back to TS 1.7?

I will check the build output and see if I can make my tsconfig match it.  I would rather be on 1.8.  But if there is another dependency besides the visual studio Type Script tools that controls what version of typescript is compiled with I would like to know it.

**Comment by mhegazy** [comment]
When uninstalling a version, there are some shared tools that are left behind. namelly the SDK.
uninstalling TS 1.7 removes the VS support. but your build tools stay around untill VS is uninstalled.

the targets and tasks has that support. 

so with that setup your MSBuild tasks know about tsconfig. removing the tsconfig.json from the project does the trick as you explained.

if you want to remove it completely from your system:
1. uninstall all TS versions you have, from Add/Remove Programs
2. delete the folder `C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\TypeScript`
3. install the version you intend, be it 1.7.6 or another.

**Comment by Vaccano** [comment]
My build had the following options (when running TS 1.7):

 `--module System --sourcemap --target ES5 --emitDecoratorMetadata --experimentalDecorators --noEmitOnError --locale en-US `

I updated my tsconfig to look like this:

    {
      "compilerOptions": {
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "locale": "en-US",
        "module": "system",
        "noEmitOnError": true,
        "sourceMap": true,
        "target": "es5"
      }
    }

And I still get the errors.  Here is a list of a few of them in case it matters (I shortened the paths):


    Error	TS2304	Cannot find name 'Promise'.	TypeScript Virtual Projects	    C:\src\entities\box\box-callback-interfaces.ts
    Error	TS2503	Cannot find namespace 'breeze'.	TypeScript Virtual Projects	C:\src\entities\box\box-type-interface.ts	
    Error	TS2503	Cannot find namespace 'breeze'.	TypeScript Virtual Projects	C:\src\entities\box\box-type.ts	
    Error	TS2503	Cannot find namespace 'breeze'.	TypeScript Virtual Projects	C:\src\entities\box\box-type.ts	
    Error	TS2307	Cannot find module 'aurelia-framework'.	TypeScript Virtual Projects	C:\src\entities\box\box.ts	
    Error	TS1219	Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option to remove this warning.	TypeScript Virtual Projects	C:\src\entities\box\box.ts
    Error	TS2307	Cannot find module 'aurelia-framework'.	TypeScript Virtual Projects	C:\src\entities\entity-factory.ts

**Comment by mhegazy** [comment]
Can you share the project?

**Comment by Vaccano** [comment]
@mhegazy Alas, no.  But I think I may have gotten it working.

What was messing me up is that with TS 1.7 it would only compile what was in my project.  With TS 1.8 it looks at the disk, not the project to see what to compile.  I had to add some excludes and remove an erroneous exclude that I had in my tsconfig to get it looking at the right stuff..  

Once I got that all figured, it started working correctly again.

Thank you for your help!
